core/main.go

```diff
@@ -246,6 +246,31 @@ func main() {
 
 	secretServiceBaseURL := fmt.Sprintf("https://%s:%s/", config.SecretService.Server, config.SecretService.Port)
 
+	hasMongoCredentails, err := credentialInStore(config, secretServiceBaseURL, config.SecretService.MongodbinitSecretPath, client)
+	if err != nil {
+		lc.Error(fmt.Sprintf("Failed to check if the MongoDB initlization parameters are in the secret store: %s", err.Error()))
+		os.Exit(1)
+	}
+
+	if hasMongoCredentails == true {
+		lc.Info("MongoDB initialization parameters are in the secret store already. Skip creating the credentials.")
+	} else {
+		initMongoDBCredentials(config, secretServiceBaseURL, client)
+	}
+
+	hasCoredataCredentials, err := coredataCredentialInStore(config, secretServiceBaseURL, config.SecretService.CoredataSecretPath, client)
+
```

Comment:
I see a credentials map in your future -- keys will be service keys and credentials will be the values. Similar to how the SMA [manages](https://github.com/edgexfoundry/edgex-go/blob/2f19d8d60cdac3e29037629d412b4bc21b4f67a4/internal/system/agent/init.go#L234) its GeneralClient instances.